packages/suite/src/components/suite/ModalSwitcher/ForegroundAppModal.tsx

```diff
@@ -11,10 +11,15 @@ import { Version } from '@suite-views/version';
 import { SwitchDevice } from '@suite-components/SwitchDevice';
 import type { ForegroundAppRoute } from '@suite-types';
 
+// would not work if defined directly in the switch
+const FirmwareType = () => <Firmware switchType />;
```

Comment:
What about passing the `switchType` prop like `cancelable` is passed further down? Or you like this approach more?

Reply:
I prefer this way because (1) the prop is only used in the `Firmware` component so I don't want to clutter the general `ForegroundAppModal`, and (2) the `app` is already checked once in the `getForegroundApp`. But I think either way is fine...